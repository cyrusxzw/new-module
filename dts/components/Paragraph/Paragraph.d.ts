import { FC } from 'react';
import type { ParagraphProps, ParagraphSetProps } from './Paragraph.types';
declare const Paragraph: FC<ParagraphProps>;
declare const ParagraphSet: FC<ParagraphSetProps>;
export { Paragraph, Paragraph as P, ParagraphSet };
