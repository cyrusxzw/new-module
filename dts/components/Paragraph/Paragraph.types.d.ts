import type { Themes } from '~/types';
declare type ParagraphProps = {
  className?: string;
  hasSerifFont?: boolean;
  isFlush?: boolean;
  isHero?: boolean;
  isLarge?: boolean;
  theme?: Themes;
};
declare type ParagraphSetProps = {
  className?: string;
  isLarge?: boolean;
  theme?: Themes;
};
export type { ParagraphProps, ParagraphSetProps };
