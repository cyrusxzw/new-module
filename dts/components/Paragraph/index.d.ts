import { Paragraph, ParagraphSet } from './Paragraph';
export { Paragraph, Paragraph as P, ParagraphSet };
