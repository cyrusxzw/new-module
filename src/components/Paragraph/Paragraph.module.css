@import 'styles/index';

.paragraph {
  margin: 0;
  margin-bottom: 30px;
  font-size: rem(14px);
  line-height: 1.6;
}

.base {
  @extend .paragraph;

  font-family: var(--font-family-sans-serif-regular);
}

.large {
  &,
  & p {
    font-size: rem(16px);
    line-height: 1.7;
  }
}

.serifFont {
  @extend %zaph-serif-typography;
}

.hero {
  font-size: rem(24px);
  line-height: 1.33;

  &.serifFont {
    font-size: rem(25px);
  }

  @media (--viewport-md) {
    font-size: rem(26px);

    &.serifFont {
      font-size: rem(30px);
    }
  }

  @media (--viewport-lg) {
    font-size: rem(30px);

    &.serifFont {
      font-size: rem(30px);
    }
  }
}

.flush {
  margin-bottom: 10px;
}

.dark {
  &,
  & p {
    @extend %font-theme-dark-copy;
  }
}

.light {
  &,
  & p {
    @extend %font-theme-light-copy;
  }
}

.set {
  p {
    @extend .paragraph;
  }

  &.large {
    &,
    & p {
      font-size: rem(16px);
      line-height: 1.7;
    }
  }
}
