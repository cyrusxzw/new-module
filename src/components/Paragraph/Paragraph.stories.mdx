import { Meta, Story, Canvas, ArgsTable } from '@storybook/addon-docs/blocks';
import * as knobs from '@storybook/addon-knobs';
import { Paragraph, P, ParagraphSet } from './Paragraph';

<Meta title="Components/Paragraph" component={Paragraph} />

# Paragraph

<Canvas>
  <Story name="Base component">
    <Paragraph
      theme={knobs.select('theme', ['dark', 'light'], 'dark', 'Presentation')}
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam id erat
      est. Ut convallis, neque id semper fermentum, urna risus gravida nibh, vel
      molestie sapien leo ac tortor. Nunc dignissim quis arcu a lobortis.
      Phasellus tempus consequat erat, at egestas elit pulvinar lacinia. Nullam
      pretium, risus a dictum rhoncus, ex libero cursus erat, dignissim
      malesuada tortor justo quis enim. Nulla pellentesque sit amet ante et
      molestie. Mauris interdum odio at mauris pulvinar, non placerat nulla
      tempus.
    </Paragraph>
  </Story>
</Canvas>

## API

<ArgsTable of={Paragraph} />

## Variations

### ParagraphSet

<Canvas>
  <Story name="ParagraphSet component">
    <ParagraphSet
      theme={knobs.select('theme', ['dark', 'light'], 'dark', 'Presentation')}
    >
      <Paragraph>
        Vivamus sollicitudin ac ligula nec convallis. Etiam varius faucibus
        libero id congue. In sit amet dignissim magna. Nullam auctor id nulla
        non sagittis. Quisque non enim efficitur quam mattis semper a nec arcu.
        In elementum placerat sapien ut malesuada. Mauris venenatis vitae orci
        et tempor. Praesent at purus eget turpis eleifend rutrum.
      </Paragraph>
      <Paragraph>
        Suspendisse elit mauris, pretium malesuada elit vel, eleifend pretium
        eros. Fusce ultricies sapien vitae diam vehicula dapibus. Nullam et
        felis sem. Phasellus posuere porta varius. Nam sit amet fringilla felis.
        Donec egestas, velit eget semper vehicula, mauris nisl sodales libero, a
        rhoncus ipsum felis ac magna. Sed vehicula pretium tortor eget mollis.
        Maecenas quis erat hendrerit, aliquam augue et, tincidunt enim.
      </Paragraph>
    </ParagraphSet>
  </Story>
</Canvas>
