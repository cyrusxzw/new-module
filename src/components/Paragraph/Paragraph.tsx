import React, { FC } from 'react';
import cx from 'classnames';
import type { ParagraphProps, ParagraphSetProps } from './Paragraph.types';
import styles from './Paragraph.module.css';

const Paragraph: FC<ParagraphProps> = ({
  children,
  className,
  hasSerifFont = false,
  isFlush = false,
  isHero = false,
  isLarge = false,
  theme = 'dark',
}) => {
  const classSet = cx(
    styles.base,
    styles[theme],
    { [styles.flush]: isFlush },
    { [styles.large]: isLarge },
    { [styles.hero]: isHero },
    { [styles.serifFont]: hasSerifFont },
    className,
  );

  return <p className={classSet}>{children}</p>;
};

const ParagraphSet: FC<ParagraphSetProps> = ({
  children,
  className,
  isLarge = false,
  theme = 'dark',
}) => {
  const classSet = cx(
    styles.set,
    styles[theme],
    { [styles.large]: isLarge },
    className,
  );

  return <div className={classSet}>{children}</div>;
};

export { Paragraph, Paragraph as P, ParagraphSet };
