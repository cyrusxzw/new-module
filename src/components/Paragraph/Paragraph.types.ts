import type { Themes } from '~/types';

type ParagraphProps = {
  className?: string;
  hasSerifFont?: boolean;
  isFlush?: boolean;
  isHero?: boolean;
  isLarge?: boolean;
  theme?: Themes;
};

type ParagraphSetProps = {
  className?: string;
  isLarge?: boolean;
  theme?: Themes;
};

export type { ParagraphProps, ParagraphSetProps };
