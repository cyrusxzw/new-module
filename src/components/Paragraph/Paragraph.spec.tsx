import React from 'react';
import { render } from '@testing-library/react';
import { Paragraph, ParagraphSet } from './Paragraph';

describe('<Paragraph />', () => {
  it('should be defined', () => {
    expect(Paragraph).toBeDefined();
  });

  it('should render base component correctly', () => {
    const { container } = render(
      <Paragraph>
        A lightweight, vitamin C-rich layering serum that soothes, replenishes
        and balances skin with its complex blend of anti-oxidant, hydrating
        conditioning ingredients.
      </Paragraph>,
    );

    expect(container).toMatchSnapshot();
  });
});

describe('<ParagraphSet />', () => {
  it('renders a given collection of `p` tags', () => {
    const { container } = render(
      <ParagraphSet>
        <p>A lightweight, vitamin C-rich layering serum that soothes</p>
        <p>
          Replenishes and balances skin with its complex blend of anti-oxidant,
          hydrating and conditioning ingredients.
        </p>
      </ParagraphSet>,
    );

    expect(container).toMatchSnapshot();
  });
});
