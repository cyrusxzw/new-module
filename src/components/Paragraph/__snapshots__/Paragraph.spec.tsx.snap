// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Paragraph /> should render base component correctly 1`] = `
<div>
  <p
    class="base dark"
  >
    A lightweight, vitamin C-rich layering serum that soothes, replenishes and balances skin with its complex blend of anti-oxidant, hydrating conditioning ingredients.
  </p>
</div>
`;

exports[`<ParagraphSet /> renders a given collection of \`p\` tags 1`] = `
<div>
  <div
    class="set dark"
  >
    <p>
      A lightweight, vitamin C-rich layering serum that soothes
    </p>
    <p>
      Replenishes and balances skin with its complex blend of anti-oxidant, hydrating and conditioning ingredients.
    </p>
  </div>
</div>
`;
